src/compiler/scala/tools/nsc/typechecker/SyntheticMethods.scala

```diff
@@ -359,13 +360,14 @@ trait SyntheticMethods extends ast.TreeDSL {
         for ((m, impl) <- methods ; if shouldGenerate(m)) yield impl()
       }
       def extras = {
-        if (needsReadResolve) {
+        if (needsModuleSerializationProxy) {
           // Aha, I finally decoded the original comment.
           // This method should be generated as private, but apparently if it is, then
           // it is name mangled afterward.  (Wonder why that is.) So it's only protected.
```

Comment:
This comment seems out of date - this method is (and was) `private`


is there some corresponding "don't mangle this name" setting somewhere